processor/src/test/java/org/mapstruct/ap/test/bugs/_2983/Issue2983Mapper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._2983;
```

Comment:
We use the `org.mapstruct.ap.test.bugs` package to put tests for different bugs. This issue is not a bug, but rather an enhancements to the new feature. Let's move the test to the `org.mapstruct.ap.test.annotatewith` and more precisely let's add new test cases in `AnnotateWithTest`. Ideally we should have a dedicated mapper per mapping method which means:

* `AnnotateBeanMappingMethodMapper`
* `AnnotateIterableMappingMethodMapper`
* `AnnotateMapMappingMethodMapper`
* `AnnotateStreamMappingMethodMapper`
* `AnnotateValueMappingMethodMapper`

Have also a look at how we are testing the generated code in the different tests in `AnnotateWithTest`